### PR TITLE
Wrong character power level warning

### PR DESCRIPTION
**Pull request by nev-r**
image

image

### CODE REVIEW

**Comment by nev-r** [comment]
why does this hit so many files? i made the repeated yellow ⚠️ into a shared component

File: src/app/gear-power/GearPower.tsx [hunk 1]
```diff
@@ -83,6 +91,19 @@ export default function GearPower() {
             );
           })}
         </div>
+        {selectedStore.stats.maxGearPower?.statProblems?.notOnStore && (
+          <div className={styles.notes}>
+            <AlertIcon /> {t('Loadouts.OnWrongCharacterWarning')}
+            {maxItemsSearchString && (
+              <>
+                <br />
+                <br />
```
Comment (bhollis) [inline src/app/gear-power/GearPower.tsx:100]: Generally prefer spacing with CSS

File: src/app/inventory/d2-stores.ts [hunk 2]
```diff
@@ -449,27 +443,44 @@ function updateBasePower(
...
-  hasClassified: boolean,
   characterProgress: DestinyCharacterProgressionComponent | undefined,
   bonusPowerProgressionHash: number | undefined
 ) {
   if (!store.isVault) {
     const def = defs.Stat.get(StatHashes.Power);
     const { equippable, unrestricted } = maxLightItemSet(allItems, store);
+
+    // ALL WEAPONS count toward your drops. armor on another character doesn't count.
+    // (maybe just because it's on a different class? who knows. can't test.)
```
Comment (bhollis) [inline src/app/inventory/d2-stores.ts:454]: I bet we could get someone with multiples of the same class to help test

File: src/app/inventory/d2-stores.ts [hunk 3]
```diff
@@ -449,27 +443,44 @@ function updateBasePower(
...
+
+    statProblems.notEquippable = unrestrictedMaxGearPower !== equippableMaxGearPower;
+    statProblems.notOnStore = dropPowerLevel !== unrestrictedMaxGearPower;
+
+    statProblems.hasClassified = allItems.some(
+      (i) =>
+        i.classified &&
+        (i.location.inWeapons ||
+          i.location.inArmor ||
+          (isD1Item(i) && i.bucket.hash === BucketHashes.Ghost))
```
Comment (bhollis) [inline src/app/inventory/d2-stores.ts:475]: Maybe we can just check for power?

**Review by bhollis — APPROVED** [review]
Very helpful. I wonder if we should file a bug since I don't think anyone knows this.

File: src/app/inventory/d2-stores.ts (hunk 3 above)
Comment (nev-r) [reply src/app/inventory/d2-stores.ts:475]: like i.power? sure yeah

Comment (bhollis) [reply src/app/inventory/d2-stores.ts:475]: I *think* that distinguishes between actual power and like... sparrow speed. I'm not sure.

**Pull request merged**